specs/Standards/MultiToken/README.md

```diff
@@ -0,0 +1,539 @@
+    pub expires_at: Option<String>, // ISO 8601 datetime when token expires
+    pub starts_at: Option<String>, // ISO 8601 datetime when token starts being valid
+    pub updated_at: Option<String>, // ISO 8601 datetime when token was last updated
+    pub extra: Option<String>, // anything extra the NFT wants to store on-chain. Can be stringified JSON.
+    pub reference: Option<String>, // URL to an off-chain JSON file with more info.
+    pub reference_hash: Option<Base64VecU8>, // Base64-encoded sha256 hash of JSON from reference field. Required if `reference` is included.
+}
+
+/// Offers details on the  metadata.
+pub trait MultiTokenMetadataProvider {
+    fn mt_metadata(&self, token_id: TokenId) -> MultiTokenMetadata;
+}
+```
+# Drawbacks
+[drawbacks]: #drawbacks
+Doing this adds another spec and codebase to the standards. It could be seen that we could leave this to developers to implement custom solutions and have them create a token that just uses FT and NFT together. There is some additional complexity
+in ux, when considering batch size request and gas limitations, that might trip some developers up. 
+
+# Rationale and alternatives
+[rationale-and-alternatives]: #rationale-and-alternatives
+The rationale for this design, is that we want to support developers that need batch requests. In many ecosystems, a single NFT or 
+series of NFT are not sufficient for representing and managing potentially 100s of tokens. Managing cross contract calls and the token contracts themselves become a huge burden on the developer. This will reduce the complexity, and allow developers to easily bridge over to other contracts like ERC-1155 on other chains that allow for a this style of representation.
+
+In the design phase it was considered to simply rely on the underlying implementations of fungible tokens and non-fungible tokens
+to be the scope of interaction with the chain. Doing this we would have tied the implementations to the bounds of FT and NFT. By loosing this up a bit, we are able to be a bit more flexible in what's possible.
+
+Not doing this means we really won't have a great way of supporting use cases where developers need to represent and manage
+large varying quantities and types of tokens. They will all have to implement it on their own, and that would make for a fragmented
+and weak ecosystem. Where every developer would not be able to reliably trade these assets.
+
+# Unresolved questions
+[unresolved-questions]: #unresolved-questions
+The unresolved questions, are really what type of metadata , is required for this spec?
+- We decided that having metadata on chain is good, and that at later time we can have a spec for extra data if needed
+
+Can we represent events in this spec, which would improve the ecosystem quite a bit? If we represent events what should those events be?
+- We decided events aren't here yet in the ecosystem, in the way we' like them to be so they are't apart of this standard
```

Comment:
This is a top concern currently; as it currently stands the indexer has no insights into third-party minting and as such we can't easily show tokens that were minted to a specific user via a callback in the wallet.  I think it would be a _really good_ idea to _define logs that should be generated as part of this standard_; those logs will be critical for the indexer to be able to identify who received which tokens as part of a complex transaction involving cross-contract callbacks; if you leave these out of the spec and a bunch of people implement contracts based on this, you will forever have a subset of tokens that are 'invisible' to the indexer (and thus, to the wallet) due to this. cc: @frol @mikedotexe

Reply:
We were really bullish on a log representation as well initially, but we wanted to have a useful standard prior to this. It's part of the reason why we even supply in the spec. That's because we don't have great binary events, that would be used in doing nice aggregations. The aggregations would allow more visibility into what's happening. I think https://gov.near.org/t/ethereum-like-logs-on-near/3509 . Has alot of the thinking around events. 

Otherwise we'd probably replicate the event structure from erc-1155 with transfer 0 being the equivalent initial minting event. We haven't had a long discussion about this because of the emerging events structure that might land sometime in the future. 

We also thought that logs might be insufficient as is to achieve what we'd like . If you have thoughts on a transitional system I'm sure we'd be down to explore that :D